### DISCUSSION TITLE
unimath.py token decimal use

### DISCUSSION THREAD

**Discussion by Tiergarten** [Jeiwan/uniswapv3-book#51] [Milestone 1. First Swap]
When doing a Swap for USDC to ETH and a swap for ETH to USDC in the book (Milestone 1: First Swap and `unimath.py`) you use the same `decimals` multiplier for both `amount_in` calculations:

```
# Swap USDC for ETH
amount_in = 42 * eth

print(f"\nSelling {amount_in/eth} USDC")

...


# Swap ETH for USDC
amount_in = 0.01337 * eth

print(f"\nSelling {amount_in/eth} ETH")


```

Is this correct? I'd expect when sending in USDC it needs to me multiplied by the `decimals` from the USDC ERC-20 contract (I think this is 6 decimal places).

Thanks,
Matt

**Comment by Mattiatore** [comment]
I think the tokens are minted and do not correspond to the standard ETH and USDC, when we deploy those tokens we specify the number of decimals (which is the same in our deployment parameters).

**Reply by Jeiwan** [reply]
Correct. I use 18 decimals in both ETH and USDC for simplicity.
